optuna/structs.py

```diff
@@ -353,14 +353,40 @@ class TrialPruned(exceptions.TrialPruned):
 
     Example:
 
-        .. code::
-
-            >>> def objective(trial):
-            >>>     ...
-            >>>     for step in range(n_train_iter):
-            >>>         ...
-            >>>         if trial.should_prune():
-            >>>             raise TrailPruned()
+        .. testsetup::
+
+            import numpy as np
+            from sklearn.model_selection import train_test_split
+
+            np.random.seed(seed=0)
+            X = np.random.randn(200).reshape(-1, 1)
+            y = np.where(X[:, 0] < 0.5, 0, 1)
+            X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
+            classes = np.unique(y)
+
+        .. testcode::
+
+            import optuna
+            from sklearn.linear_model import SGDClassifier
+
+            def objective(trial):
+                alpha = trial.suggest_uniform('alpha', 0.0, 1.0)
+                clf = SGDClassifier(alpha=alpha)
+                n_train_iter = 100
+
+                for step in range(n_train_iter):
+                    clf.partial_fit(X_train, y_train, classes=classes)
+
+                    intermediate_value = 1.0 - clf.score(X_test, y_test)
+                    trial.report(intermediate_value, step)
+
+                    if trial.should_prune():
+                        raise optuna.structs.TrialPruned()
+
+                return 1.0 - clf.score(X_test, y_test)
+
+            study = optuna.create_study(pruner=optuna.pruners.MedianPruner())
+            study.optimize(objective, n_trials=20)
```

Comment:
Please leave the example as it is because `optuna.structs.TrialPruned` is deprecated. I think we can remove this example because it increases synchronization cost if we keep both of the examples.